project_work.py

```diff
@@ -0,0 +1,250 @@
+for brand, sales in brand_dict.items():
+    if sales < other_category_cutoff:
+        others += sales / num_deals * 100
+    else:
+        percentage_brand_dict.update({brand: sales / num_deals * 100})
+percentage_brand_dict.update({"Others": others})
+percentage_brand_dict_sorted = sorted(percentage_brand_dict.items(), key=lambda item: item[1], reverse=True)
+for brand_and_percentage in percentage_brand_dict_sorted:
+    print(f"{brand_and_percentage[0]} : {round(brand_and_percentage[1], 2)}%")
+print()
+print("=" * 19)
+
+# data sets for entry into the graphs
+states = list(set(state_list))
+sales = list(state_dict.keys())
+
+
+
+# creates the bar chart for question 2
+fig = plt.figure(1)
+plt.bar(states, sales)
+plt.xlabel("State", fontsize= 16)
+plt.ylabel("Amount of sale", fontsize = 16)
+plt.title("Amount of sale in different states", fontsize = 16)
+
+# we need the code for the question 3 graph here
+
+
+# we need the code to graph the question 4 pie chart here
+plt.figure(3)
+labels = []
+sizes = []
+for brand_percentage in percentage_brand_dict_sorted:       #Separates the dictionary into two lists to use as labels and sizes for the pie graph.
+    labels.append(brand_percentage[0]) 
+    sizes.append(brand_percentage[1])
+colors = ['b', 'g', 'r', 'c', 'm', 'y']
+number = 0
+for num in range(len(percentage_brand_dict_sorted)):    #Helps assign colors to all elements of chart
+    if number == len(percentage_brand_dict_sorted) - 1:
+        number = 0
+    colors.append(colors[number])
+    number += 1
+plt.pie(sizes, labels = labels, colors = colors, autopct='%1.2f%%')
+
+# we need the code to graph the last line graph for question 5 here
+car_data_file = open("2019_car_sale.csv")
+car_data_reader = csv.reader(car_data_file)
+date_state_sale_dict = {'Texas':[],'Ohio':[],'California':[],'Nevada':[],'Florida':[]}
+header = True
+for deal in car_data_reader:        #Creates dictionary for the function that will makes the y-points for each state
+    if header:
+        header = False
+        continue    
+    date = deal[1].split("/")[0]
+    state = deal[3]
+    sale = int(deal[5].replace(",", ""))
+    date_state_sale_dict[state].append([date, sale])
+    
+x_points = ['Janurary','February','March','April','May','June','July','August','September','October','November','December']
+texas_y_points = amount_sale_month_state(date_state_sale_dict, 'Texas')
+cali_y_points = amount_sale_month_state(date_state_sale_dict, 'California')
+florida_y_points = amount_sale_month_state(date_state_sale_dict, 'Florida')
+ohio_y_points = amount_sale_month_state(date_state_sale_dict, 'Ohio')
+nevada_y_points = amount_sale_month_state(date_state_sale_dict, 'Nevada')
+car_data_file.close()           #All y-points created
+
+plt.xlabel('Months')        #Customization
+plt.ylabel('Amount of Sale')
+plt.title('Amount of Sale in Different Months in Each State')
+
+plt.figure(4)       #Plots the points with respective legend

```

Comment:
```suggestion
plt.figure(4, figsize = (12,5))       #Plots the points with respective legend

```
This will make the figure bigger so that the month ticks are not overlapping